rust/connlib/tunnel/src/tests.rs

```diff
@@ -1399,7 +1451,12 @@ impl ReferenceState {
             ResourceDst::Dns(domain) => {
                 tracing::Span::current().record("dst", tracing::field::display(domain));
 
-                if self.client_dns_records.contains_key(domain) && src.originates_from_client() {
+                if self.client_dns_records.get(domain).is_some_and(|ips| {
+                    ips.iter().any(|(ip, status)| {
+                        matches!(status, IpStatus::Queried) && ip.is_ipv4() == src.is_ipv4()
```

Comment:
I know there probably won't be anything past IPv6 for decades but something like `ip.ip_version() == src.ip_version()` would make it more obvious that we're just checking for the versions to match, and it will also be true for 6 == 6.